File: REST/core/e1c.py
import requests, sys
import pprint

'''
Use the /lookup endpoint to find IRAK4
Then find the associated sequence.
'''

server = "http://rest.ensembl.org"
lookup_endpoint = "/lookup/symbol/human/{}"
sequence_endpoint = "/sequence/id/{}"

def fetch_endpoint(server, request, content_type='application/json'):
    """
    Fetch an endpoint from the server, allow overriding of default content-type
    """
    r = requests.get(server+request, headers={ "Content-Type" : content_type})

    if not r.ok:
        r.raise_for_status()
        sys.exit()

    if content_type == 'application/json':
        return r.json()
    else:
        return r.text

if __name__ == "__main__":

    print "Fetching gene IRAK4"

    # Fetch the record for the gene IRAK4
    gene = fetch_endpoint(server, lookup_endpoint.format('IRAK4'))

    # Extract the stable id from the dictionary and print
    print "Stable id: {}".format(gene['id'])

    # Fetch the sequence in FASTA format
    sequence = fetch_endpoint(server, sequence_endpoint.format(gene['id']), 'text/x-fasta')

    # Print the sequence we received
    print sequence

File: REST/core/e1c.pl
#!/bin/env perl

# Use the /lookup endpoint to find IRAK4
# Then find associated sequence

use strict;
use warnings;

use HTTP::Tiny;
use JSON;
use Data::Dumper;
$Data::Dumper::Terse = 1;
$Data::Dumper::Indent = 1;

my $server = "http://rest.ensembl.org";
my $lookup_endpoint = "/lookup/symbol/human/%s";
my $sequence_endpoint = "/sequence/id/%s";

my $http = HTTP::Tiny->new();

# Fetch an endpoint from the serverm allow overriding of the default content type
sub fetch_endpoint {
    my ($request, $content_type) = @_;
    $content_type ||= 'application/json';

    my $response = $http->get($request,
                              { headers => { 'Content-type' => $content_type } } );

    die "Failed!\n" unless $response->{success};

    if($content_type eq 'application/json') {
	return decode_json($response->{content});
    } else {
	return $response->{content};
    }
}

print "Fetching gene IRAK4\n";

my $url = $server . sprintf($lookup_endpoint, 'IRAK4');
my $gene = fetch_endpoint($url);

print "Stable id: " . $gene->{id} . "\n";

$url = $server . sprintf($sequence_endpoint, $gene->{id});
my $sequence = fetch_endpoint($url, 'text/x-fasta');
print $sequence;
